test/e2e/app-dir/ppr-errors/ppr-errors.test.ts

```diff
@@ -66,24 +66,6 @@ describe('ppr build errors', () => {
             'Error occurred prerendering page "/no-suspense-boundary"'
           )
         })
-
-        it('should fail the build & surface any errors that were thrown by user code', async () => {
-          // in the case of catching a postpone and throwing a new error, we log the error that the user threw to help with debugging
-          expect(stderr).toContain(
-            'Prerendering /no-suspense-boundary-re-throwing-error needs to partially bail out because something dynamic was used. '
-          )
-          expect(stderr).toContain(
-            'The following error was thrown during build, and may help identify the source of the issue:'
-          )
-          expect(stderr).toContain(
-            "Error: Throwing a new error from 'no-suspense-boundary-re-throwing-error'"
-          )
-
-          // the regular pre-render error should not be thrown as well, as we've already logged a more specific error
-          expect(stderr).not.toContain(
-            'Error occurred prerendering page "/no-suspense-boundary-re-throwing-error"'
-          )
-        })
```

Comment:
This is technically incorrect, as this is race dependent. The error that reaches the top level first should be the one that the user has to manage, not the deeper error. Once the user had fixed this specific error, they would instead see the error about the missing postpone data which could be handled.

Reply:
The only bummer here is that given this code:

```tsx
try {
  cookies();
} catch (err) {
  throw new Error(
    "Throwing a new error from 'no-suspense-boundary-re-throwing-error'",
  );
}
```
The error will now be:
```
Error occurred prerendering page "/no-suspense-boundary-re-throwing-error". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Throwing a new error from 'no-suspense-boundary-re-throwing-error'
```

Which seems confusing, since as a user I wouldn't expect `cookies()` to throw and might be lead me down the wrong path looking at other unrelated parts of my code. This is why previously it would say:

```
⨯ Prerendering /no-suspense-boundary-re-throwing-error needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error
 ⚠ The following error was thrown during build, and may help identify the source of the issue:
 ⨯ Error: Throwing a new error from 'no-suspense-boundary-re-throwing-error'
``` 

It seems like we still should handle this case somehow, but perhaps out of scope for this PR.  🤔